src/hooks/usePushTopics.js

```
@@ -0,0 +1,40 @@
+import useNotificationsStore from '@/hooks/useNotificationsStore';
+import { requestPermissionAndToken, subscribeTopics, unsubscribeTopics } from '@/lib/push';
+import { track } from '@/lib/analytics';
+
+export default function usePushTopics() {
+  const { user } = useAuth();
+  const { allCommunities } = useCommunitySwitcher();
+  const { pushEnabled } = useNotificationsStore();
+  const tokenRef = useRef(null);
+
+  useEffect(() => {
+    let cancelled = false;
```

ESLint

[error] 'cancelled' is assigned a value but never used. Allowed unused vars must match /^[A-Z_]/u.

(no-unused-vars)